File: 프로그래머스/N개의 최소공배수.py
def gcd(a, b):
    while b > 0:
        a, b = b, a % b
    return a

def solution(arr):
    # a가 b보다 크게 넣어줘야 하므로 먼저 정렬해주기
    arr.sort()
    arr.reverse()
    answer = arr[0]
    
    for i in range(1, len(arr)):
        answer = (answer * arr[i]) // gcd(answer, arr[i])
    
    return answer

File: 프로그래머스/N개의 최소공배수.java
class Solution {
    private int gcd(int a, int b) {
        while (b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }

    public int solution(int[] arr) {
        int result = 1;
        for (int i = 0; i < arr.length; i++)
            result = arr[i] * result / gcd(arr[i], result);
        return result;
    }
}